src/main/java/racingcar/Racing.java

```diff
@@ -0,0 +1,85 @@
+package racingcar;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class Racing {
+
+    public static final int POWER_BOUND = 9;
+
+    private static final Random random = new Random();
+    private static List<Car> cars;
+    private static int raceCount;
+    private static List<List<Car>> raceResults;
+
+    public static void main(String[] args) {
+        Racing.setRace(InputView.askCarCount(null), InputView.askRaceCount(null));
+        Racing.startRace();
+        OutputView.endRace();
+    }
+
+    public static void setRace(int carCount, int raceCount) {
+        setCars(carCount);
+        setRaceCount(raceCount);
+        setRaceResults();
+    }
+
+    private static void setRaceResults() {
+        raceResults = new ArrayList<>();
+    }
+
+    private static void setRaceCount(int raceCount) {
+        Racing.raceCount = raceCount;
+    }
+
+    private static void setCars(int carCount) {
+        cars = new ArrayList<>();
+        for (int i = 0; i < carCount; i++) {
+            cars.add(new Car());
+        }
+    }
+
+    public static void startRace() {
+        for (int r = 0; r < raceCount; r++) {
+            raceCars();
+            saveRaceResult();
+        }
+    }
+
+    private static void raceCars() {
+        int size = cars.size();
+        for (int i = 0; i < size; i++) {
+            raceCar(cars.get(i), getNextInt());
+        }
+    }
+
+    private static void saveRaceResult() {
+        List<Car> result = new ArrayList<>();
+        for (Car car : cars) {
+            result.add(car.clone());
+        }
+        raceResults.add(result);
+    }
+
+    private static void raceCar(Car car, int power) {
+        car.move(power);
+    }
+
+    private static int getNextInt() {
+        return random.nextInt(POWER_BOUND + 1);
+    }
+
+    public static List<Car> getCars() {
+        return cars;
```

Comment:
자동차 참가자 명단은 중요하니까요 😄 
참조한 객체 쪽에서 변경이 어렵도록 불변객체로 반환되면 더 좋을 것 같습니다 🙇 

이 부분 한번 참고해보시면 좋을 것 같습니다 🙏 
> https://velog.io/@kyle/%EB%B6%88%EB%B3%80-%EA%B0%9D%EC%B2%B4%EB%9E%80-Java-Immutable-Object

Reply:
> 관련해 일급컬렉션 클래스에서 해당 멤버변수 컬렉션을 수정 불가하도록(add, remove 등 불가) 하는건 가능했는데, 해당 컬렉션에 들어있는 객체를 꺼내서 해당 객체의 상태변경을 요청하면 (public 메소드 호출) 정상 동작하더라구요.
> 
> 일급컬렉션이 보장하는 불변의 범위는 해당 클래스의 멤버변수 까지인 걸까요?? 아니면 제가 잘못 사용하고 있는걸까요??

관련해, CarsTest에 '자동차_멤버변수_값_변경불가' 라는 테스트는 실패상태로 커밋해두었습니다.
`() -> assertThatExceptionOfType(UnsupportedOperationException.class)
                .isThrownBy(() -> result.get(0).move()`

Reply:
> 관련해 일급컬렉션 클래스에서 해당 멤버변수 컬렉션을 수정 불가하도록(add, remove 등 불가) 하는건 가능했는데, 해당 컬렉션에 들어있는 객체를 꺼내서 해당 객체의 상태변경을 요청하면 (public 메소드 호출) 정상 동작하더라구요.
일급컬렉션이 보장하는 불변의 범위는 해당 클래스의 멤버변수 까지인 걸까요?? 아니면 제가 잘못 사용하고 있는걸까요??

한 depth 까지 더 들어가셨군요 그 부분까지 생각하셨다니 너무 좋네요 💯 

맞습니다 성홍님이 말씀하신게 진정한 의미의 불변객체라고 할 수 있겠습니다 
반환한 `List`를 변경하는 `operation`은 막혀 있지만 사실 그 안에 들어있는 객체들의 
`reference`는 그대로 가지고 있기 때문에 접근이 가능한 상태입니다 🙇 

그래서 그 부분까지 방지하려면 List내부에 있는 객체들도 새로 `clone`해서 아예 새 `List`를 만들어야 
원본을 보존하는 의미의 불변을 가져갈 수 있습니다 😄 

이와 비슷한 구현 방식이 Java 내부에도 많이 있는데요 
`BigDecimal`이라는 큰 수 연산에 사용하는 객체인데요 `operation`의 반환 값을 대 부분 새 객체로 반환하도록 해서
원본 값을 보존하는 형식으로 되어있습니다 
```java
/**
     * Translates a {@code long} unscaled value and an
     * {@code int} scale into a {@code BigDecimal}.  This
     * {@literal "static factory method"} is provided in preference to
     * a ({@code long}, {@code int}) constructor because it
     * allows for reuse of frequently used {@code BigDecimal} values..
     *
     * @param unscaledVal unscaled value of the {@code BigDecimal}.
     * @param scale scale of the {@code BigDecimal}.
     * @return a {@code BigDecimal} whose value is
     *         <tt>(unscaledVal &times; 10<sup>-scale</sup>)</tt>.
     */
    public static BigDecimal valueOf(long unscaledVal, int scale) {
        if (scale == 0)
            return valueOf(unscaledVal);
        else if (unscaledVal == 0) {
            return zeroValueOf(scale);
        }
        return new BigDecimal(unscaledVal == INFLATED ?
                              INFLATED_BIGINT : null,
                              unscaledVal, scale, 0);
    }

```
개선 까지는 필요 없을 것 같고 
관련해서 한번 참고만 해주시면 좋을 것 같습니다 😄